app/controllers/register/beneficiaries/charityOrTrust/charity/AddressInTheUkYesNoController.scala

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers.register.beneficiaries.charityOrTrust.charity
+
+import controllers.actions._
+import controllers.actions.register._
+import forms.YesNoFormProvider
+import javax.inject.Inject
+import models._
+import navigation.Navigator
+import pages.register.beneficiaries.charityortrust.charity.{AddressInTheUkYesNoPage, CharityNamePage}
+import play.api.data.Form
+import play.api.i18n._
+import play.api.mvc._
+import repositories.RegistrationsRepository
+import uk.gov.hmrc.play.bootstrap.controller.FrontendBaseController
+import views.html.register.beneficiaries.charityortrust.charity.AddressInTheUkYesNoView
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class AddressInTheUkYesNoController @Inject()(
+                                               override val messagesApi: MessagesApi,
+                                               registrationsRepository: RegistrationsRepository,
+                                               navigator: Navigator,
+                                               identify: RegistrationIdentifierAction,
+                                               getData: DraftIdRetrievalActionProvider,
+                                               requireData: RegistrationDataRequiredAction,
+                                               requiredAnswer: RequiredAnswerActionProvider,
+                                               formProvider: YesNoFormProvider,
+                                               val controllerComponents: MessagesControllerComponents,
+                                               view: AddressInTheUkYesNoView
+                                             )(implicit ec: ExecutionContext) extends FrontendBaseController with I18nSupport {
+
+  private val form: Form[Boolean] = formProvider.withPrefix("charity.addressInTheUkYesNo")
+
+  private def actions(draftId: String, index: Int) =
+    identify andThen
+      getData(draftId) andThen
+      requireData andThen
+      requiredAnswer(RequiredAnswer(CharityNamePage(index), routes.CharityNameController.onPageLoad(NormalMode, index, draftId)))
+
+  def onPageLoad(mode: Mode, index: Int, draftId: String): Action[AnyContent] = actions(draftId, index) {
+    implicit request =>
+
+      val charityName = request.userAnswers.get(CharityNamePage(index)).get
```

Comment:
Have we not got an action that we can include to pull out the charity name, similar to how we've done it for other beneficiaries.

This would mean you don't need to do `requiredAnswer(RequiredAnswer(CharityNamePage(index), routes.CharityNameController.onPageLoad(NormalMode, index, draftId)))`
and ` val charityName = request.userAnswers.get(CharityNamePage(index)).get` and it would be available in the `implicit request`.

See https://github.com/hmrc/register-trust-beneficiary-frontend/blob/c6616a5fe4c066b6654c587bfcc7a4cc7b3c82e4/app/controllers/actions/register/company/NameRequiredAction.scala for an example